aim/db/hashtree_db_listener.py

```diff
@@ -36,44 +36,64 @@ def on_commit(self, session, added, updated, deleted):
         # Segregate updates by tenant
         updates_by_tenant = {}
         all_updates = [added, updated, deleted]
+        conf = tree_model.CONFIG_TREE
+        monitor = tree_model.MONITORED_TREE
+        oper = tree_model.OPERATIONAL_TREE
         for idx in range(len(all_updates)):
+            tree_index = 0 if idx < 2 else 1
             for res in all_updates[idx]:
                 key = self.tt_maker.get_tenant_key(res)
                 if not key:
                     continue
-                updates_by_tenant.setdefault(key, ([], [], [], []))
-                updates_by_tenant[key][
-                    (0 if idx < 2 else 1) +
-                    (2 if isinstance(res, aim_status.OperationalResource)
-                     else 0)].append(res)
+                updates_by_tenant.setdefault(
+                    key, {conf: ([], []), monitor: ([], []), oper: ([], [])})
+                if isinstance(res, aim_status.OperationalResource):
+                    # Operational Tree
+                    updates_by_tenant[key][oper][tree_index].append(res)
+                elif getattr(res, 'monitored', None):
+                    # Monitored Tree
+                    updates_by_tenant[key][monitor][tree_index].append(res)
```

Comment:
I believe it is required to support some external connectivity configurations. In any case, the design for monitored objects doesn't seem to account for the possibility that we may want to modify relations of pre-existing objects, and that looks like a major weakness to me.